include/HelicityFormalism.h

```diff
@@ -98,7 +98,11 @@ class HelicityFormalism : public SpinAmplitudeCache
     /// \param L orbital angular momentum
     /// \param two_S 2 * the total spin angular momentum
     virtual std::shared_ptr<SpinAmplitude> create(unsigned two_J, const SpinVector& two_j, unsigned l, unsigned two_s) const override
-    { return (two_J == 0) ? unit(two_J, two_j, l, two_s) : std::shared_ptr<SpinAmplitude>(new HelicitySpinAmplitude(two_J, two_j, l, two_s)); }
+    {
+        if (two_J == 0 and std::all_of(two_j.begin(), two_j.end(), [](int i){ return i == 0; }))
```

Comment:
Alternatively, `UnitSpinAmplitude` can take an argument to set the constant it returns (other than `1`)---perhaps it should then be renamed `ConstantSpinAmplitude`.

Such a `ConstantSpinAmplitude` is needed when `J == 0`;
the create function should throw if `L != S or is_odd(L + two_j[0] + two_j[1])`.
Else, the coefficient is:
```cpp
auto two_m = projections(*std::min_element(two_j.begin(), two_j.end()));
auto C = pow_negative_one(L) / sqrt(2 * L + 1)
     * std::accumulate(two_m.begin(), two_m.end(), 0.,
                      [&](double& c; int m)
                    {return c += ClebschGordon::coefficient(two_j[0], m, two_j[1], -m, L, 0);});
```